imports/plugins/core/ui/client/components/badge/badge.js

```diff
@@ -4,52 +4,26 @@ import classnames from "classnames/dedupe";
 import { Components, registerComponent } from "@reactioncommerce/reaction-components";
 
 class Badge extends Component {
-  constructor(props) {
-    super(props);
-
-    this.state = {
-      tooltipOpen: false
-    };
-
-    this.handleButtonMouseOver = this.handleButtonMouseOver.bind(this);
-    this.handleButtonMouseOut = this.handleButtonMouseOut.bind(this);
-  }
+  state = {
+    tooltipOpen: false
+  };
 
   get isTooltipOpen() {
     return this.state.tooltipOpen;
   }
 
-  handleButtonMouseOver() {
+  handleMouseOver = () => {
```

Comment:
I am admittedly new to ES# syntax, but this technique yields a method where `this` is set properly?
at first glance, it looks like `this` would refer to the class `Badge` and not an instance of `Badge`

Reply:
it refers to the instance of the class